src/commands/containers/viewContainerLogs.ts

```diff
@@ -15,7 +16,15 @@ export async function viewContainerLogs(context: IActionContext, node?: Containe
         });
     }
 
+    const terminalCommand = await selectTemplate(
+        context,
+        'attach',
+        `${node.containerName} ${node.fullTag}`,
+        undefined,
+        { 'containerId': node.containerId }
```

Comment:
There's a strong coupling between the variables provided for replacement and those documented/expected for each command.  Having the clients of `selectTemplate()` pass in the variables by name allows for easy misspelling or accidental omission, as well as requires every use of the template to pass in the (hopefully) same dictionary.

Why not create functions for each command (e.g. `selectLogsTemplate(containerId: string): Promise<string>`) that wraps `selectTemplate()`, allowing easy reuse, keeps the translation to variables closer to their substitution (to minimize the chance of misspelling), as well as enforces passing the expected variable values.

Reply:
Sure; though I think that `resolveVariables` should keep the behavior it has of accepting a dictionary. Would these `selectLogsTemplate` etc. methods be in this selectTemplate.ts file?

Reply:
Agreed on `resolveVariables()` itself accepting a dictionary.  I think it'd be fine for `selectLogsTemplate()` and company to be in the same file.  (You could just keep `selectTemplate()` as non-exported to prevent nefarious use outside of the file.)